.github/workflows/update-dependencies.yml

```
@@ -0,0 +1,76 @@
+---
+name: Update dependencies
+
+on:
+  schedule:
+    # Run daily (Monday-Friday) at 12:15 UTC (08:15 EDT/07:15 EST)
+    - cron: "15 12 * * 1-5"
+  workflow_dispatch:
+
+# This workflow needs to leverage a GitHub Personal Access Token (PAT) in order
+# to open the pull request. While the GitHub Actions token _could_ create the
+# pull request if the `permissions` block were edited to allow that, the created
+# PR would not actually trigger any status checks, which would be mostly useless.
+# Therefore a PAT is used. Theoretically, any of the mechanisms listed in the
+# peter-evans/create-pull-request Action's README could be used (and we may
+# eventually switch to using the app).
+permissions:
+  contents: read
+
+jobs:
+  npm:
+    name: NPM Dependencies
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          # The specific version of node itself isn't especially important
+          # to us.
+          node-version: "lts/*"
+          cache: npm
+
+      - name: Install latest npm and and npm-check-updates
+        run: npm install -g npm@latest npm-check-updates@latest
+
+      - name: Update project dependencies
+        run: |-
+          cd end-to-end-tests
+          ncu --deep --upgrade --target minor
+          ncu --deep --upgrade
+          cd ..
+
+      - name: Install updated dependencies
+        run: |-
+          cd end-to-end-tests
+          npm install
+          npm update
+          cd ..
+
+      - name: Login as the automation app
+        # This Action generates a token from the GitHub App and provides it as
+        # an output. It _does_ register that token as a secret so that it will be
+        # filtered from log output automatically
+        id: generate-token
+        uses: tibdex/github-app-token@586e1a624db6a5a4ac2c53daeeded60c5e3d50fe
+        with:
+          app_id: ${{ secrets.APP_ID }}
+          private_key: ${{ secrets.APP_PRIVATE_KEY }}
+
+      - name: Create update pull request
+        uses: peter-evans/create-pull-request@v4
+        with:
+          commit-message: Update NPM dependencies
+          title: Update NPM dependencies
+          body: >-
+            This was generated by the "${{ github.workflow }}" workflow, primarily
+            by running `node-check-updates` and `npm install`.
```

```suggestion
            by running `npm-check-updates` and `npm install`.
```

I understand this is probably incorrect in the original workflow too. Please open a PR to correct that as well.